### PR TITLE
docs: Updating Responsive Styles Support Section of V8 Guide

### PR DESCRIPTION
**Pull request by josh-bagwell**
## Summary

Fixes: #1788  

Adding responsive style support documentation to the v8 upgrade guide. 

## Release Category
Documentation

---

## Checklist

- [ ] MDX documentation adheres to Canvas Kit's [standard MDX template](https://github.com/Workday/canvas-kit/discussions/1131)

## For the Reviewer

- [ ] PR title is short and descriptive
- [ ] PR summary describes the change (Fixes/Resolves linked correctly)
- [ ] PR Release Notes describes additional information useful to call out in a release message or removed if not applicable
- [ ] Breaking Changes provides useful information to upgrade to this code or removed if not applicable

## Where Should the Reviewer Start?

`/modules/docs/mdx/8.0-UPGRADE-GUIDE.mdx`

## Areas for Feedback? (optional)

- [ ] Documentation

### CODE REVIEW

File: modules/docs/mdx/8.0-UPGRADE-GUIDE.mdx [hunk 1]
```diff
@@ -178,7 +178,76 @@ const Example = styled('div')(
 
 ### Responsive Styles
 
-> Coming Soon (https://github.com/Workday/canvas-kit/pull/1779)
+The following Canvas Kit responsive utilities will be released as part of v8:
+
+- **ResponsiveContextProvider**: This is a React Context Provider that will take a width and an
+  optional theme and allow components to subscribe to context changes via the `useResponsiveContext`
```
Comment (mannycarrera4) [inline modules/docs/mdx/8.0-UPGRADE-GUIDE.mdx:184]:
```suggestion
  optional theme prop and allow components to subscribe to context changes via the `useResponsiveContext`
```

File: modules/docs/mdx/8.0-UPGRADE-GUIDE.mdx [hunk 2]
```diff
@@ -178,7 +178,76 @@ const Example = styled('div')(
...
+  hook.
+
+- **useResponsiveContext Hook**: This hook subscribes to the `ResponsiveContextProvider` and
+  provides helpful utilities to determine whether the current width is within a particular theme
+  breakpoint. Developers can use these utilities when creating container-based responsive styles.
```
Comment (mannycarrera4) [inline modules/docs/mdx/8.0-UPGRADE-GUIDE.mdx:189]: should we reference what those  default theme breakpoints are?

File: modules/docs/mdx/8.0-UPGRADE-GUIDE.mdx (hunk 1 above)
Comment (josh-bagwell) [reply modules/docs/mdx/8.0-UPGRADE-GUIDE.mdx:184]: Updated

File: modules/docs/mdx/8.0-UPGRADE-GUIDE.mdx (hunk 2 above)
Comment (josh-bagwell) [reply modules/docs/mdx/8.0-UPGRADE-GUIDE.mdx:189]: Yes we should. I'll update

File: modules/docs/mdx/8.0-UPGRADE-GUIDE.mdx [hunk 3]
```diff
@@ -178,7 +178,77 @@ const Example = styled('div')(
...
+The following Canvas Kit responsive utilities will be released as part of v8:
+
+- **ResponsiveContextProvider**: This is a React Context Provider that will take a width and an
+  optional theme prop and allow components to subscribe to context changes via the
+  `useResponsiveContext` hook.
+
+- **useResponsiveContext Hook**: This hook subscribes to the `ResponsiveContextProvider` and
+  provides helpful utilities to determine whether the current width is within a particular theme
+  breakpoint (`Small: 320px, Medium: 768px, Large: 1024px, Extra Large: 1440px`). Developers can use
+  these utilities when creating container-based responsive styles.
```
Comment (jamesfan) [inline modules/docs/mdx/8.0-UPGRADE-GUIDE.mdx:190]: Removing an unnecessary "Hook". I agree with Manny that providing more context on breakpoints is a good idea; even better than listing them out would be to link to the breakpoints table in the Responsive Styling doc which is more explicit about when each size is applied based on the start and end of the px ranges.

I find a single value to be insufficient to understand when a particular size is applied whereas a range is more clear (i.e., `Small (s) is applied from 320px - 767px` is better than `small: 320px`).

```suggestion
- **useResponsiveContext**: This hook subscribes to the `ResponsiveContextProvider` and provides
  helpful utilities to determine whether the current width is within a particular
  [theme breakpoint](/getting-started/for-developers/resources/responsive-styling/#canvas-kit-breakpoint-sizes).
  Developers can use these utilities when creating container-based responsive styles.
```

Note that although this link _won't_ work in Storybook due to the `#`, it _will_ work on the Canvas Site (more context [here](https://github.com/Workday/canvas-kit/pull/1827#discussion_r988387128)).

Alternatively, we could reproduce the whole passage on Canvas Kit Breakpoint Sizes from your Responsive Styling here in the Upgrade Guide...

Screen Shot 2022-10-19 at 3 15 03 PM

... but I'm not sure that's necessary.

File: modules/docs/mdx/8.0-UPGRADE-GUIDE.mdx [hunk 4]
```diff
@@ -178,7 +178,77 @@ const Example = styled('div')(
...
+
+- **useResponsiveContainerStyles**: A hook that allows developers to create container-based
+  responsive styles using style objects. These are net-new utilities that developers can opt-in to
+  start using, but no action is required to uptake this change.
```
Comment (jamesfan) [inline modules/docs/mdx/8.0-UPGRADE-GUIDE.mdx:194]: The net-new bit doesn't need to be here in the Upgrade Guide.
```suggestion
- **useResponsiveContainerStyles**: A hook that allows developers to create container-based
  responsive styles using style objects.
```

File: modules/docs/mdx/8.0-UPGRADE-GUIDE.mdx [hunk 5]
```diff
@@ -178,7 +178,77 @@ const Example = styled('div')(
...
+    <Box ref={ref}>
+      <Flex {...containerResponsiveStyles.flex}>
+        <Box {...containerResponsiveStyles.box}>Hello World</Box>
+      </Flex>
+    </Box>
+  </ResponsiveContextProvider>
+);
+```
+
+This update is specific to Canvas Kit and will not impact existing designs.
```
Comment (jamesfan) [inline modules/docs/mdx/8.0-UPGRADE-GUIDE.mdx:251]: This also doesn't need to be in the Upgrade Guide.
```suggestion
```

**Review by jamesfan — COMMENTED** [review]
@josh-bagwell Thanks for putting this together! I suggested a few changes below to clarify some concepts and to tailor the content given the more developer-centric nature of the Upgrade Guide vs. the wider audience for the official announcement.

File: modules/docs/mdx/8.0-UPGRADE-GUIDE.mdx [hunk 6]
```diff
@@ -178,7 +178,77 @@ const Example = styled('div')(
 
 ### Responsive Styles
 
-> Coming Soon (https://github.com/Workday/canvas-kit/pull/1779)
+The following Canvas Kit responsive utilities will be released as part of v8:
```
[the remark this answers is not in the document]
Comment (josh-bagwell) [reply modules/docs/mdx/8.0-UPGRADE-GUIDE.mdx:181]: This makes a lot of sense. I like the link to the Responsive.stories

File: modules/docs/mdx/8.0-UPGRADE-GUIDE.mdx [hunk 7]
```diff
@@ -178,7 +178,77 @@ const Example = styled('div')(
...
+
+- **ResponsiveContextProvider**: This is a React Context Provider that will take a width and an
+  optional theme prop and allow components to subscribe to context changes via the
```
[the remark this answers is not in the document]
Comment (josh-bagwell) [reply modules/docs/mdx/8.0-UPGRADE-GUIDE.mdx:184]: Yeah I was wondering if I should use backticks in that particular part, thanks for clarifying!

File: modules/docs/mdx/8.0-UPGRADE-GUIDE.mdx (hunk 4 above)
Comment (josh-bagwell) [reply modules/docs/mdx/8.0-UPGRADE-GUIDE.mdx:194]: Updated

File: modules/docs/mdx/8.0-UPGRADE-GUIDE.mdx (hunk 5 above)
Comment (josh-bagwell) [reply modules/docs/mdx/8.0-UPGRADE-GUIDE.mdx:251]: Updated and thanks for letting me know on this!

**Review by jamesfan — APPROVED** [review]
Looks good!

**Pull request merged**